address.js

```diff
@@ -125,24 +124,33 @@ proto.extract = function(fieldName, regexes) {
   }
 
   // iterate over the unit regexes and test them against the various parts
-  for (rgxIdx = 0; rgxIdx < regexes.length; rgxIdx++) {
-    for (ii = this.parts.length; ii >= 0; ii-- ) {
+  for (ii = this.parts.length-1; ii >= 0; ii--) {
+    for (rgxIdx = 0; rgxIdx < regexes.length; rgxIdx++) {
+
+      // skip fields which have already been parsed
+      if (this[fieldName]){ continue; }
+
+      // execute regex against part
       match = regexes[rgxIdx].exec(this.parts[ii]);
 
       // if we have a match, then process
       if (match) {
-        // if we have a 2nd capture group, then replace the item with
-        // the text of that group
         if (match[2]) {
+          // if we have a 2nd capture group, then replace the item with	
+          // the text of that group
           this.parts.splice(ii, 1, match[2]);
-        }
-        // otherwise, just remove the element
-        else {
+        } else {
+          // otherwise, just remove the element from parts
           this.parts.splice(ii, 1);
         } // if..else
 
-        value = lookups[rgxIdx] || match[1];
-      } else if (fieldName === 'state' && value === undefined) {
+        // set the field
+        this[fieldName] = lookups[rgxIdx] || match[1];
+      } // if
```

Comment:
Probably drop this `// if` comment.  It was a bad pattern (added little value) I was in at the time of writing this library.   Then the `else if` block can be more obviously associated with this block for the sake of code readability.

Reply:
I'd be happy to remove those :)

It's not a pattern I'm familiar with but i usually try to honour the original formatting in external contributions.

I'll go ahead and remove all those from this file.

Reply:
> It's not a pattern I'm familiar with but i usually try to honour the original formatting in external contributions.

And it's appreciated, but I'm happy to see them go now.

> I'll go ahead and remove all those from this file.

Thanks.

Reply:
resolved via https://github.com/DamonOehlman/addressit/pull/25/commits/97f16989477bac7bed80606d7cfe3e0fb95f349f